clustered/client/src/main/java/org/ehcache/clustered/client/internal/loaderwriter/writebehind/ClusteredWriteBehindStore.java

```diff
@@ -0,0 +1,385 @@
+
+      return resolvedChain.getResolvedResult(key) != null;
+    } catch (Exception re) {
+      throw handleException(re);
+    }
+  }
+
+  @Override
+  protected V silentRemove(K key, V value) throws StoreAccessException {
+    try {
+      ConditionalRemoveOperation<K, V> operation = new ConditionalRemoveOperation<>(key, value, timeSource.getTimeMillis());
+      ByteBuffer payload = codec.encode(operation);
+      long extractedKey = extractLongKey(key);
+      Chain chain = storeProxy.getAndAppend(extractedKey, payload);
+      ResolvedChain<K, V> resolvedChain = resolver.resolve(chain, key, timeSource.getTimeMillis());
+
+      Result<K, V> result = resolvedChain.getResolvedResult(key);
+      return result == null ? null : result.getValue();
+    } catch (Exception re) {
+      throw handleException(re);
+    }
+  }
+
+  @Override
+  protected V silentReplace(K key, V value) throws StoreAccessException {
+    try {
+      ReplaceOperation<K, V> operation = new ReplaceOperation<>(key, value, timeSource.getTimeMillis());
+      ByteBuffer payload = codec.encode(operation);
+      long extractedKey = extractLongKey(key);
+      Chain chain = storeProxy.getAndAppend(extractedKey, payload);
+      ResolvedChain<K, V> resolvedChain = resolver.resolve(chain, key, timeSource.getTimeMillis());
+
+      Result<K, V> result = resolvedChain.getResolvedResult(key);
+      return result == null ? null : result.getValue();
+    } catch (Exception re) {
+      throw handleException(re);
+    }
+  }
+
+  protected V silentReplace(K key, V oldValue, V newValue) throws StoreAccessException {
+    try {
+      ConditionalReplaceOperation<K, V> operation = new ConditionalReplaceOperation<>(key, oldValue, newValue, timeSource
+        .getTimeMillis());
+      ByteBuffer payload = codec.encode(operation);
+      long extractedKey = extractLongKey(key);
+      Chain chain = storeProxy.getAndAppend(extractedKey, payload);
+      ResolvedChain<K, V> resolvedChain = resolver.resolve(chain, key, timeSource.getTimeMillis());
+
+      Result<K, V> result = resolvedChain.getResolvedResult(key);
+      return result == null ? null : result.getValue();
+    } catch (Exception re) {
+      throw handleException(re);
+    }
+  }
+
+  public class WriteBehindServerCallback implements ServerStoreProxy.ServerCallback {
+
+    private final ServerStoreProxy.ServerCallback delegate;
+
+    WriteBehindServerCallback(ServerStoreProxy.ServerCallback delegate) {
+      this.delegate = delegate;
+    }
+
+    @Override
+    public void onInvalidateHash(long hash) {
+      this.delegate.onInvalidateHash(hash);
+    }
+
+    @Override
+    public void onInvalidateAll() {
+      this.delegate.onInvalidateAll();
+    }
+
+    @Override
+    public Chain compact(Chain chain) {
+      return this.delegate.compact(chain);
+    }
+
+    @Override
+    public Chain compact(Chain chain, long hash) {
+      clusteredWriteBehind.flushWriteBehindQueue(chain, hash);
+      return null;
```

Comment:
Why null? And not the compacted chain?

Reply:
compacted chain is not available at this point, SOR will be updated in flushWriteBehindQueue and then replaceAthead will be called with the resulting compacted chain.